File: game/entity.go
package game

type Entity struct {
	ID         int32
	Game       *Game
	Components []Component
}

func NewEntity(game *Game, comps []Component) Entity {
	return Entity{-1, game, comps}
}

func (e *Entity) GetComponent(compType string) int {
	for i, c := range e.Components {
		if c.GetType() == compType {
			return i
		}
	}
	return -1
}

File: game/entity.lua
local entity = {}

function entity:new(game, comps)
    local instance = {
        game = game,
        components = comps
    }
    setmetatable(instance, entity)
    return instance
end

function entity:AddComponent(comp)
    entity.components[comp.name] = comp
end

function entity:RemoveComponent(comp)
    entity.components[comp.name] = nil
end

return entity